js/src/Ice/RouterInfo.js

```diff
@@ -47,41 +46,35 @@ export class RouterInfo {
         return this._router;
     }
 
-    getClientEndpoints() {
-        const promise = new Promise();
-        if (this._clientEndpoints !== null) {
-            promise.resolve(this._clientEndpoints);
-        } else {
-            this._router
-                .getClientProxy()
-                .then(result => this.setClientEndpoints(result[0], result[1] !== undefined ? result[1] : true, promise))
-                .catch(promise.reject);
+    async getClientEndpoints() {
+        if (this._clientEndpoints === null) {
+            const [clientProxy, hasRoutingTable] = await this._router.getClientProxy();
+            this._hasRoutingTable = hasRoutingTable ?? true;
+            this._clientEndpoints =
+                clientProxy === null
+                    ? this._router._getReference().getEndpoints()
+                    : clientProxy._getReference().getEndpoints();
```

Comment:
Move the implementation of `setClientEndpoints` here, as it is only used in this place.